### PR TITLE
Aquifer edge cases

### PR DESCRIPTION
**Pull request by silochad**

### CODE REVIEW

File: test/Aquifer.t.sol [hunk 1]
```diff
@@ -120,47 +120,75 @@ contract AquiferTest is TestHelper {
...
+            "",
+            initFunctionCall,
+            bytes32(0)
+        );
     }
 
-    function test_bore_fail_message() public {
-        initFunctionCall = abi.encodeWithSignature("initMessage()");
+    /// @dev Revert if the Well implementation doesn't have the provided init function.
+    /// FIXME: does fallback behavior affect this?
```
Comment (k-xo) [inline test/Aquifer.t.sol:142]: Is this in reference to if a Well implements a `fallback` function? Could make a mock well that has such a behaviour, but, shouldn't it be okay, as all state changing functions are `nonReentrant`

Comment (silochad) [reply test/Aquifer.t.sol:142]: https://docs.soliditylang.org/en/v0.8.19/contracts.html#fallback-function

Tried this POC:

```
contract MockInitFailWell {

    function initNoMessage() external pure {
        revert();
    }

    function initMessage() external pure {
        revert("Well: fail message");
    }

    fallback (bytes calldata) external returns (bytes memory) {}

}
```

```
[FAIL. Reason: Call reverted as expected, but without data] test_bore_expectRevert_missingInitFunction() (gas: 61162)
Traces:
  [61162] AquiferTest::test_bore_expectRevert_missingInitFunction()
    ├─ [0] VM::expectRevert(Aquifer: well init)
    │   └─ ← ()
    ├─ [50610] Aquifer::boreWell(MockInitFailWell: [0x1d1499e622D69689cdf9004d05Ec547d650Ff211], 0x, 0x1dcc85ae, 0x0000000000000000000000000000000000000000000000000000000000000000)
    │   ├─ [8828] → new <Unknown>@0x4f81992FCe2E1846dD528eC0102e6eE1f61ed3e2
    │   │   └─ ← 44 bytes of code
    │   ├─ [2791] 0x4f81992FCe2E1846dD528eC0102e6eE1f61ed3e2::doesNotExist()
    │   │   ├─ [132] MockInitFailWell::doesNotExist() [delegatecall]
    │   │   │   └─ ← ()
    │   │   └─ ← ()
    │   ├─ [291] 0x4f81992FCe2E1846dD528eC0102e6eE1f61ed3e2::aquifer() [staticcall]
    │   │   ├─ [132] MockInitFailWell::aquifer() [delegatecall]
    │   │   │   └─ ← ()
    │   │   └─ ← ()
    │   └─ ← "EvmError: Revert"
    └─ ← "Call reverted as expected, but without data"
    ```
    
This reverts because the mock doesn't define an `aquifer()` function, i.e. this line returned `success = true`:
```
            (bool success, bytes memory returnData) = well.call(initFunctionCall);```

Comment (silochad) [reply test/Aquifer.t.sol:142]: Not sure there's much we can do about this behavior; if a dev sets up a Well with a fallback function, and then incorrectly encodes the name of the init function to call during `boreWell()`, not sure there's any way for us to know that things didn't get executed as expected.

I'll change this from FIXME to NOTE for now. If you agree with the above logic, let's move fwd with just the note

Comment (k-xo) [reply test/Aquifer.t.sol]: yep I think the note and the logic makes sense! LGTM

**Review by k-xo — APPROVED** [review]

**Review by k-xo — APPROVED** [review]
